FF1Blazorizer/Tabs/FunTab.razor

```diff
@@ -37,30 +38,40 @@
 
 	protected override async void OnInitialized()
 	{
-		Flags.Preferences = await LocalStorage.GetItemAsync<Preferences>("preferences") ?? new Preferences();
-		if (Flags.Preferences.SpriteSheet != null) {
-		  SpriteSheetMessage = "Using custom sprites.";
-		}
+		Flags.Preferences = new Preferences();
+		await LoadPreferences();
 	}
 
-	async Task SavePreferences()
+	void SavePreferences()
 	{
-		await LocalStorage.SetItemAsync("preferences", Flags.Preferences);
+	        string jsonprefs = JsonSerializer.Serialize(Flags.Preferences);
+		JSRuntime.InvokeVoidAsync("setFFRPreferences", "preferences", jsonprefs);
 		SetStatusMessage.Invoke("Preferences saved.");
 	}
 
-	private string SpriteSheetMessage = "Load custom player sprites:";
-
-		async Task OnSpriteSheetChanged(ChangeEventArgs e)
-		{
-		      Console.WriteLine("OnSpriteSheetChanged");
-	              Flags.SpriteSheet = await JSRuntime.InvokeAsync<string>("handleFileSelect", "spriteSheetInput");
-		      SpriteSheetMessage = "Using custom sprites.";
+	async Task LoadPreferences() {
+		string prefs = await JSRuntime.InvokeAsync<string>("getFFRPreferences", "preferences");
+		if (prefs == "" || prefs == null) return;
+		Flags.Preferences = JsonSerializer.Deserialize<FF1Lib.Preferences>(prefs);
+		if (Flags.Preferences.SpriteSheet != null) {
+		  SpriteSheetMessage = "Using custom sprites.";
 		}
+		StateHasChanged();
+	}

```

Comment:
Im concerned about how this might work with changing formats for the preferences. This will be more of an issue with presets, but is still something to consider with preferences.

Can you confirm that this works with:
1. blank preferences
2. renamed preferences
Im not sure if it will throw if `prefs` does not match `FF1Lib.Preferences`

Reply:
@mhn65536 already did a fix to handle incompatible preset flags.
This does something a little simpler, incompatible preferences fields should be ignored by a no-op error handler, and it merges the newly saved presets with the existing presets to maintain forward compatibility if you save preferences on an older site, it won't lose keys from preferences on a newer site.